### PR TITLE
Port ocaml/ocaml#11727 and ocaml/ocaml#11732

### PR DESCRIPTION
**Pull request by stedolan**
Two small type system fixes for first-class module issues in 4.14 (reviewed upstream)

### CODE REVIEW

File: ocaml/typing/typemod.ml [hunk 1]
```diff
@@ -2142,7 +2142,7 @@ and package_constraints env loc mty constrs =
 
 let modtype_of_package env loc p fl =
   package_constraints env loc (Mty_ident p)
```
Comment (antalsz) [inline ocaml/typing/typemod.ml:2144]: Please bring in the comment here, too!

**Review by antalsz — CHANGES_REQUESTED** [review]
You missed a comment we should bring in, and you could also bring in the `Changes` update if you wanted; happy to merge after you do #1 (and #2 if you want to)

File: ocaml/typing/typemod.ml (hunk 1 above)
Comment (stedolan) [reply ocaml/typing/typemod.ml:2146]: Done

**Review by antalsz — APPROVED** [review]

**Comment by stedolan** [comment]
I generally don't bother porting edits to Changes to flambda-backend, since it's not read by users of this repo and it's a constant source of merge conflicts if we try to update it.

**Review by lpw25 — APPROVED** [review]

**Pull request merged**